### PR TITLE
[release-4.13] OCPBUGS-38455: Update opentelemetry to mitigate CVE

### PR DESCRIPTION
**Pull request by apoorvajagtap**
Bumps (primarily):

go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc to v0.46.0 [OCPBUGS-38455]
go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp to v0.44.0 [OCPBUGS-38456]
& related go.opentelemetry.io dependencies

### CODE REVIEW

**Review by avinal — APPROVED** [review]
LGTM

**Comment by apoorvajagtap** [comment]
/assign @ingvagabund 
for review & approval

File: go.mod [hunk 1]
```diff
@@ -18,35 +18,40 @@ require (
 	k8s.io/klog/v2 v2.80.1
 )
 
+require (
+	google.golang.org/genproto/googleapis/api v0.0.0-20230822172742-b8732ec3820d // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20230822172742-b8732ec3820d // indirect
```
Comment (ingvagabund) [inline go.mod:23]: One nit: can you put these two lines under the `require` block below?

**Pull request merged**